File: testsuite/basic/src/test/java/io/smallrye/faulttolerance/reuse/async/completionstage/fallback/guard/MyFaultTolerance.java
package io.smallrye.faulttolerance.reuse.async.completionstage.fallback.guard;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Produces;

import io.smallrye.common.annotation.Identifier;
import io.smallrye.faulttolerance.api.Guard;

@ApplicationScoped
public class MyFaultTolerance {
    @Produces
    @Identifier("my-fault-tolerance")
    public static final Guard GUARD = Guard.create()
            .withRetry().maxRetries(2).done()
            .build();
}

File: testsuite/basic/src/test/kotlin/io/smallrye/faulttolerance/kotlin/reuse/fallback/guard/MyFaultTolerance.kt
package io.smallrye.faulttolerance.kotlin.reuse.fallback.guard

import io.smallrye.common.annotation.Identifier
import io.smallrye.faulttolerance.api.Guard
import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.inject.Produces

@ApplicationScoped
object MyFaultTolerance {
    @Produces
    @Identifier("my-fault-tolerance")
    val GUARD = Guard.create()
        .withRetry().maxRetries(2).done()
        .build()
}
